ssl/ssl_cert.c

```diff
@@ -1323,7 +1323,7 @@ const SSL_CERT_LOOKUP *ssl_cert_lookup_by_pkey(const EVP_PKEY *pk, size_t *pidx,
         }
     }
     /* check provider-loaded pk types */
-    for (i = 0; ctx->sigalg_list_len; i++) {
+    for (i = 0; i < ctx->sigalg_list_len; i++) {
```

Comment:
I've added this fix to #26671 open against master, that will likely land first, so it'll drop out of this one at some point on a rebase or merge. This particular fix will then be back ported to 3.2 onwards.